### PR TITLE
feat(knative): make installation explicit

### LINKED ISSUE #4716: Helm installation procedure does not support Knative
### Can't install Camel-K operator using helm chart

Have 2 simple integrations

hello.groovy
```groovy
from('timer:tick?period=3000')
        .setBody().constant('Hello world from Camel K!')
        .to('log:info')
```

HttpTest.java
```java
package com.backbase.gc.mambu;
import org.apache.camel.builder.RouteBuilder;
@SuppressWarnings("unused")
public class HttpTest extends RouteBuilder {
    @Override
    public void configure() {
        from("rest://get:/test")
                .setBody().simple("Hello world from test at ${date:now:yyyy-MM-dd HH:mm:ss.SSS}")
                .log("${body}")
        ;
    }
}
```

If I install Camel-K using cli:
```shell
kamel install -n camel-k --force --olm=false \
  --global \
  --organization ecosystemsgctmp \
  --registry ecosystemsgctmp.azurecr.io \
  --registry-secret camel-k-registry-secret \
  --wait
```
everything work.

If I use helm to install Camel-K operator:

```shell
helm -n camel-k upgrade --install -f camel-k.values.yaml \
  --repo https://apache.github.io/camel-k/charts \
  camel-k \
  camel-k \
  --version 2.0.1 --wait
```
camel-k.values.yaml
```yaml
---
operator:
  global: "true"
platform:
  build:
    registry:
      organization: ecosystemsgctmp
      address: ecosystemsgctmp.azurecr.io
      secret: camel-k-registry-secret
```

### Steps to reproduce

- Kubernetes cluster: AKS v1.26.6
- istio, istio-gateway: 1.18.2
- knative: v1.11

### Install Camel-K operator using following command

```shell
helm -n camel-k upgrade --install -f camel-k.values.yaml \
  --repo https://apache.github.io/camel-k/charts \
  camel-k \
  camel-k \
  --version 2.0.1 --wait
```
camel-k.values.yaml
```yaml
---
operator:
  global: "true"
platform:
  build:
    registry:
      organization: ecosystemsgctmp
      address: ecosystemsgctmp.azurecr.io
      secret: camel-k-registry-secret
```

### Run integraion

HttpTest.java
```java
package com.backbase.gc.mambu;
import org.apache.camel.builder.RouteBuilder;
@SuppressWarnings("unused")
public class HttpTest extends RouteBuilder {
    @Override
    public void configure() {
        from("rest://get:/test")
                .setBody().simple("Hello world from test at ${date:now:yyyy-MM-dd HH:mm:ss.SSS}")
                .log("${body}")
        ;
    }
}
```
with command:
```shell
kamel run --trait knative-service.min-scale=1 HttpTest.java
```

### Relevant log output

```shell
{"level":"info","ts":1693918064.4193337,"logger":"camel-k.traits","msg":"Skipping environment variable POD_NAME (fieldRef)","trait":"container"}
{"level":"error","ts":1693918064.4319084,"msg":"Reconciler error","controller":"integration-controller","controllerGroup":"camel.apache.org","controllerKind":"Integration","Integration":{"name":"http-test","namespace":"default"},"namespace":"default","name":"http-test","reconcileID":"5349f143-288d-4f01-8df0-ee61fb41c5c0","error":"error executing post actions: error during apply resource: default/http-test: services.serving.knative.dev \"http-test\" is forbidden: User \"system:serviceaccount:camel-k:camel-k-operator\" cannot patch resource \"services\" in API group \"serving.knative.dev\" in the namespace \"default\": RBAC: clusterrole.rbac.authorization.k8s.io \"addressable-resolver\" not found","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\tsigs.k8s.io/controller-runtime@v0.13.1/pkg/internal/controller/controller.go:326\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tsigs.k8s.io/controller-runtime@v0.13.1/pkg/internal/controller/controller.go:273\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\tsigs.k8s.io/controller-runtime@v0.13.1/pkg/internal/controller/controller.go:234"}
```
```


### Camel K version

v2.0.1

### PR DESCRIPTION
**Pull request by squakez**
Instead of letting the CLI perfom the installation, we need to make it explicit in order to use the same approach for any other installation methodology.

Closes #4716

**Release Note**
```release-note
feat(knative): make installation explicit
```

### CODE REVIEW

File: config/rbac/knative/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+namespace: my-ns
+resources:
+# Required in order to perform namespace replacement below
+- operator-service-account.yaml
+- operator-role-knative.yaml
+- operator-role-binding-knative.yaml
+- operator-cluster-role-binding-custom-resource-definitions.yaml
+replacements:
```
Comment (squakez) [inline config/rbac/knative/kustomization.yaml:26]: @gansheer I think the "replacement" can be a good way to solve the plethora of "placeholder" replacement we have in kustomization. The idea is that we include the serviceAccount (altough it exists) in order to get its namespace and replace it accordingly. wdyt?

Comment (gansheer) [reply config/rbac/knative/kustomization.yaml:26]: Using `replacements` is one of my (favorite) possible option for namespace handling. Bu I need to see how it fits with all the other rbac configurations adaptations like ClusterRoleBindings and RoleBindings that reference the service.

**Review by oscerd — APPROVED** [review]

File: docs/modules/ROOT/pages/installation/knative.adoc [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+```bash
+kubectl apply -k https://github.com/apache/camel-k/tree/main/config/rbac/knative?ref=v2.1.0
+```
+
+The configuration above is using the released version `2.1.0`. You may want to use instead another version or the latest development available on `main` branch. Before continuing with the configuration, you need to make sure to know which are the namespaces on which the Operator is running and on which your Integration is going to run (they could be the same). In order to change it, you must perform some customization locally in a similar fashion:
+
+```bash
+git clone https://github.com/apache/camel-k.git
+cd camel-k/config/rbac/knative
+sed -i "s/namespace: placeholder/namespace: $OPERATOR_NAMESPACE/" operator-cluster-role-binding-custom-resource-definitions.yaml
```
Comment (gansheer) [inline docs/modules/ROOT/pages/installation/knative.adoc:25]: From what I understand of kustomize, replacing "placeholder" by "default" in the rbac yaml files. This should also works on [subjects](https://github.com/kubernetes-sigs/kustomize/blob/e7c8ed18998745680cf45efd12e5990cedcb3d39/api/filters/namespace/namespace.go#L108-L125).

Comment (squakez) [reply docs/modules/ROOT/pages/installation/knative.adoc:25]: That would be great as it simplify the procedure a lot. However I wonder how it works when the operator is installed in a global namespace. I'll make some test to see if it is the proper solution, thanks for the hints.

Comment (gansheer) [reply docs/modules/ROOT/pages/installation/knative.adoc:25]: The only issue I had was when one of the resource need to be on another namespace that the one declared with `kustomize edit set namespace $IT_NAMESPACE` (in my case I want to keep the `camel-k-operator-local-registry` RoleBinding on namespace `kube-public`).
What is the specifics of a "global" namespace ?

Comment (squakez) [reply docs/modules/ROOT/pages/installation/knative.adoc:25]: Basically the operator could be installed in any namespace, ie `global-ns`. The service account `camel-k-operator` is already existing on that namespace and we must instruct the kustomize procedure that this is the namespaced resource we want to provide as a subject for the RoleBinding. I've tried to do some test, but I could not make it work without explicitly setting the namespace replacement :/

**Pull request merged**